system-reports/datastructures/basic_report.go

```diff
@@ -11,23 +11,73 @@ import (
 	"github.com/armosec/utils-go/httputils"
 )
 
-var (
-	systemReportEndpoint = ""
+const (
+	defaultSystemReportEndpoint = "/k8s/sysreport"
 )
 
-func SetSystemReportEndpoint(endpoint string) {
-	if endpoint == "" {
-		endpoint = "/k8s/sysreport"
+type sysEndpoint struct {
+	value string
+	mu    sync.RWMutex
+}
+
+func (e *sysEndpoint) IsEmpty() bool {
+	return e.Get() == ""
+}
+
+func (e *sysEndpoint) Set(value string) {
+	e.mu.Lock()
+	e.value = value
+	e.mu.Unlock()
+}
+
+func (e *sysEndpoint) Get() string {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+	return e.value
+}
+
+// SetOrDefault sets the system report endpoint to the provided value for valid
+// inputs, or to a default value on invalid ones
+//
+// An empty input is considered invalid, and would thus be set to a default endpoint
+func (e *sysEndpoint) SetOrDefault(value string) {
+	if value == "" {
+		value = defaultSystemReportEndpoint
 	}
-	systemReportEndpoint = endpoint
+	e.Set(value)
+}
 
+// GetOrDefault returns the value of the current system report endpoint if it
+// is set. If not set, it sets the value to a default and returns the newly set
+// value
+func (e *sysEndpoint) GetOrDefault() string {
+	current := e.Get()
+	if current == "" {
+		e.SetOrDefault("")
+	}
+	return e.Get()
 }
 
+var (
+	systemReportEndpoint = &sysEndpoint{}
+)
+
+// SetSystemReportEndpoint sets the system report endpoint to the provided
+// value or a default
+//
+// Deprecated: the SetSysetmReportEndpoint shouldn’t be the setter for an
+// unexported variable. Use the sysEndpoint.SetOrDefault() method instead.
+func SetSystemReportEndpoint(endpoint string) {
+	systemReportEndpoint.SetOrDefault(endpoint)
+}
+
+// GetSystemReportEndpoint gets the system report endpoint or sets a default
+// and returns it
+//
+// Deprecated: the GetSysetmReportEndpoint shouldn’t be the setter for an
+// unexported variable. Use the sysEndpoint.GetOrDefault() method instead.
 func GetSystemReportEndpoint() string {
-	if systemReportEndpoint == "" {
-		SetSystemReportEndpoint("")
-	}
-	return systemReportEndpoint
+	return systemReportEndpoint.GetOrDefault()
 }
```

Comment:
question: can we safely remove these functions?

I have only seen these functions being used inside of the `armosec/logget-go` package. But seeing how these are exported functions, do we expect clients of `armosec/logger-go` to use it?

If not, then I would like to remove the functions and also make sure that the new concurrent system report endpoint is unexported, private to the logger package.